File: javascript/1299.Replace-Elements-with-Greatest-Element-on-Right-Side.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function (arr) {
  const n = arr.length;
  const result = new Array(n).fill(-1);

  for (let i = n - 2; i >= 0; i--) {
    result[i] = Math.max(result[i + 1], arr[i + 1]);
  }

  return result;
};

File: python/1299.Replace-Elements-with-Greatest-Element-on-Right-Side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        result = [-1] * n

        for i in range(n - 2, -1, -1):
            result[i] = max(result[i + 1], arr[i + 1])

        return result